lib/install/management/store_files.go

```diff
@@ -201,20 +214,24 @@ func (d *Dispatcher) getChildren(ds *object.Datastore, dsPath string) ([]string,
 }
 
 // getSortedChildren returns all children under datastore path in reversed order.
-func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string) ([]string, error) {
-	result, err := d.getChildren(ds, dsPath)
+func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string, matchPattern ...string) ([]string, error) {
+	result, err := d.getChildren(ds, dsPath, matchPattern...)
 	if err != nil {
 		return nil, err
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(result)))
 	return result, nil
 }
 
-func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
+func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string, matchPattern ...string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
 	defer trace.End(trace.Begin(dsPath))
 
+	// match every subfolers by default
+	if len(matchPattern) == 0 {
+		matchPattern = append(matchPattern, "*")
+	}
 	spec := types.HostDatastoreBrowserSearchSpec{
-		MatchPattern: []string{"*"},
+		MatchPattern: matchPattern,
```

Comment:
What's the result, if you list folder test1/, and exclude test1/abc/volumes?
will test1/abc returns?
I'm asking this question, because if that is returned, you're still delete test1/abc folder no matter how.